lib/ansible/modules/cloud/azure/azure_rm_cdnprofile.py

```diff
@@ -0,0 +1,293 @@
+        provisioning_state=cdnprofile.provisioning_state,
+        tags=cdnprofile.tags
+    )
+
+
+class AzureRMCdnprofile(AzureRMModuleBase):
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str'
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            ),
+            sku=dict(
+                type='str',
+                choices=['Standard_Verizon', 'Premium_Verizon', 'Custom_Verizon', 'Standard_Akamai', 'Standard_ChinaCdn']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.state = None
+        self.tags = None
+        self.sku = None
+
+        required_if = [
+            ('state', 'present', ['sku'])
+        ]
+
+        self.results = dict(changed=False)
+
+        super(AzureRMCdnprofile, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                supports_check_mode=True,
+                                                supports_tags=True,
+                                                required_if=required_if)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        to_be_updated = False
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        response = self.get_cdnprofile()
+
+        if self.state == 'present':
+
+            if not response:
+                self.log("Need to create the CDN profile")
+
+                if not self.check_mode:
+                    self.results = self.create_cdnprofile()
+                    self.log("Creation done")
+
+                self.results['changed'] = True
+                return self.results
+
+            else:
+                self.log('Results : {0}'.format(response))
+                update_tags, response['tags'] = self.update_tags(response['tags'])
+
+                if response['provisioning_state'] == "Succeeded":
+                    if update_tags:
+                        to_be_updated = True
+
+                if to_be_updated:
+                    self.log("Need to update the CDN profile")
+
+                    if not self.check_mode:
+                        self.results = self.update_cdnprofile()
+                        self.log("Update done")
+
+                    self.results['changed'] = True
+                    return self.results
+
+        elif self.state == 'absent' and response:
```

Comment:
what  if self.state == absent and null response? any message to user?

Reply:
This means user is trying to delete an non-existed profile. No msg to user, but 'changed' is False in this situation. (If deletion is successful 'changed' is True)